### PR TITLE
clients(lr): run benchmark repeatedly given special query parameter

### PR DESCRIPTION
**Pull request by paulirish**
This weird mode is for some lightrider variance work underway.

Example output: 
image

cc @adrianaixba

### CODE REVIEW

File: types/artifacts.d.ts [hunk 1]
```diff
@@ -49,6 +49,8 @@ interface UniversalBaseArtifacts {
   LighthouseRunWarnings: Array<string | IcuMessage>;
   /** The benchmark index that indicates rough device class. */
   BenchmarkIndex: number;
+  /** Many benchark indexes. Many. */
```
Comment (connorjclark) [inline types/artifacts.d.ts:52]: typo

File: types/lhr/lhr.d.ts [hunk 2]
```diff
@@ -59,6 +59,8 @@ declare module Result {
     networkUserAgent: string;
     /** The benchmark index number that indicates rough device class. */
     benchmarkIndex: number;
+    /** Many benchmark indexes */
```
Comment (connorjclark) [inline types/lhr/lhr.d.ts:62]: we seem to like periods in these comments

File: lighthouse-core/gather/gather-runner.js [hunk 3]
```diff
@@ -491,6 +491,17 @@ class GatherRunner {
       const baseArtifacts = await GatherRunner.initializeBaseArtifacts(options);
       baseArtifacts.BenchmarkIndex = await getBenchmarkIndex(driver.executionContext);
 
+      if (options.settings.__internalMegaBenchmarkIndex) {
+        const bidxRunCount = options.settings.__internalMegaBenchmarkIndex;
+        // Keep the first one in there
+        const bidxes = [baseArtifacts.BenchmarkIndex];
+        for (let i = 0; i < bidxRunCount; i++) {
```
Comment (connorjclark) [inline lighthouse-core/gather/gather-runner.js:498]: It'd be simpler if the internal setting was just minus one'd here (And we asserted it either doesn't exist or is >= 1).

File: types/lhr/settings.d.ts [hunk 4]
```diff
@@ -97,6 +97,8 @@ export type ScreenEmulationSettings = {
   precomputedLanternData?: PrecomputedLanternData | null;
   /** The budget.json object for LightWallet. */
   budgets?: Array<Budget> | null;
+  /** Run benchmarkindex this many extra times */
+  __internalMegaBenchmarkIndex?: number;
```
Comment (connorjclark) [inline types/lhr/settings.d.ts:101]: see other comment: it's less weird to define this option as "run it this many times" vs "run it this many extra times"

File: lighthouse-core/gather/gather-runner.js [hunk 5]
```diff
@@ -491,6 +491,17 @@ class GatherRunner {
       const baseArtifacts = await GatherRunner.initializeBaseArtifacts(options);
       baseArtifacts.BenchmarkIndex = await getBenchmarkIndex(driver.executionContext);
 
+      if (options.settings.__internalMegaBenchmarkIndex) {
+        const bidxRunCount = options.settings.__internalMegaBenchmarkIndex;
+        // Keep the first one in there
+        const bidxes = [baseArtifacts.BenchmarkIndex];
```
Comment (connorjclark) [inline lighthouse-core/gather/gather-runner.js:497]: nit: bidxes -> indexes?

File: clients/lightrider/lightrider-entry.js [hunk 6]
```diff
@@ -61,6 +61,11 @@ export async function runLighthouseInLR(connection, url, flags, lrOpts) {
       config.settings.onlyCategories = categoryIDs;
     }
   }
+  // Set to 0 to disable
```
Comment (connorjclark) [inline clients/lightrider/lightrider-entry.js:64]: Set what to 0? Who is this comment for?

Comment (connorjclark) [reply clients/lightrider/lightrider-entry.js:64]: Oh, I think I see. Could you instead make `megaBidx` be optional, or union with `| false` ?

File: clients/lightrider/lightrider-entry.js [hunk 7]
```diff
@@ -37,11 +37,11 @@ globalThis.Buffer = Buffer;
  * @param {Connection} connection
  * @param {string} url
  * @param {LH.Flags} flags Lighthouse flags
- * @param {{lrDevice?: 'desktop'|'mobile', categoryIDs?: Array<string>, logAssets: boolean, configOverride?: LH.Config.Json}} lrOpts Options coming from Lightrider
+ * @param {{lrDevice?: 'desktop'|'mobile', categoryIDs?: Array<string>, logAssets: boolean, configOverride?: LH.Config.Json, megaBidx: number}} lrOpts Options coming from Lightrider
```
Comment (connorjclark) [inline clients/lightrider/lightrider-entry.js:40]: rename to `collectMultipleBenchmarkIndexes` ?

**Comment by connorjclark** [comment]
I assume I can review this fine without understanding the motivation :)

File: lighthouse-core/gather/gather-runner.js (hunk 3 above)
Comment (paulirish) [reply lighthouse-core/gather/gather-runner.js:502]: as we're the only users i think this isn't a huge deal. and it'd complicate the impl

File: clients/lightrider/lightrider-entry.js (hunk 6 above)
Comment (paulirish) [reply clients/lightrider/lightrider-entry.js]: moved it to `channel='fr' && queryParam('bidx')` instead.

**Review by connorjclark — APPROVED** [review]

**Pull request merged**